File: top-k-frequent/python/topKFrequent.py
def topKFrequent(nums):
    dic =  dict()
    for num in nums:
        if num in dic:
           dic[num] += 1
        else:
            dic[num] = 1

    result = sorted(dic, key=dic.get, reverse=True)[:k] 
    return result

nums = [1,1,1,2,2,3]
k = 2
print(topKFrequent(nums))
        

File: array-hashing/top-k-frequent/java/top-k-frequent.java
public static int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer, Integer> hashmap = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int currentVal = nums[i];
            if (hashmap.containsKey(currentVal)) {
                int prevCounter = hashmap.get(currentVal);
                int newCounter = prevCounter + 1;
                hashmap.put(currentVal, newCounter);
            } else {
                hashmap.put(currentVal, 1);
            }
        }


        int[] result = hashmap
                .entrySet()
                .stream()
                .sorted(Map.Entry.<Integer, Integer>comparingByValue().reversed())
                .limit(k)
                .mapToInt(Map.Entry::getKey)
                .toArray();

        return result;
    }